_posts/2024-04-23-secure-blog-migration.md

```diff
@@ -0,0 +1,151 @@
+which can help stop malicious attacks.
+
+Think of security headers as little helpers that work behind the scenes to 
+keep you and your information safe while you're browsing the web.
+
+### The consequences of no security headers
+#### Security
+- Cross-site scriptiong (XSS) attacks
+  - Involves injection of malicious scripts into your website.
+  - This can be done by an attacker using MIME Sniffing.
+- Clickjacking
+  - Involves hiding malicious elements under innocent-looking ones to deceive users into clicking on them.
+
+#### SEO
+Why invest time in creating great content if it isn't safe to interact with? An unsecured site puts users' online safety at risk and can undermine your SEO efforts. 
+Google may label your site as unsafe, which can damage your reputation and result in less clicks a month.
+
+#### Visitors
+Visitors might lose trust and leave. This could harm your brand's reputation, 
+as users may see your site as less trustworthy, affecting how they perceive your business overall.
+
+## Our solution
+To address the security concerns, we made the decision to move our web hosting from GitHub
+Pages to a platform that allows us to configure security headers. This led us to migrate
+our blog to an Amazon S3 bucket within our AWS environment. Here we utilize AWS CloudFront
+to set up and enforce the necessary security headers.
+
+In addition, we've implemented an AWS WAF (Web Application Firewall) into our infrastructure
+to further enhance our website's security posture. Given our existing experience with AWS 
+through previous project, we decided to leverage the platform for its robust security features.
+
+By resolving this security issue, we seized the opportunity to migrate our blog 
+repository from GitHub Organization to GitHub Enterprise. 
+However, the old repository remains in place to preserve the original 
+domain name and redirect users to our new website.
+
+## Action plan details
+We've used Terraform as Infrastructure as a Code combined with GitHub Actions to deploy all our resources.
+1. AWS WAF
+* Implemented **Rate limiting** to protect against DDoS Attacks.
+* Integrated Managed AWS Rules:
+  * **Common Rules** to protect against exploitation of a wide range of vulnerabilities including those described in OWASP publications.
+  * **Known Bad Input Rules** to block requests patterns that are known to be invalid and are associated with  exploitation or discovery of vulnerabilities.
+  * **Amazon Ip reputation List**  to block sources associated with bots or other threats.
+
+2. AWS S3 Bucket
+* We created an S3 bucket that is fully private and only accessible by Cloudfront. 
+
+3. AWS ACM & Route 53
+* Created a subdomain for our blog using Route 53. 
+* Generated a certificate via AWS ACM to ensure secure connections for our visitors. 
+This certificate will be integrated into our CloudFront configuration.
+
+4. AWS Cloudfront
+* Initiated configuration of the origin, directing to our S3 bucket.
```

Comment:
What is meant here? This is unclear for me.

Reply:
The origins settings of CloudFront. 
![image](https://github.com/Ordina-Group/JWorks-blog/assets/112859645/5d7cd6c6-fb49-41c1-a905-957a295af759)

But if it's already unclear for you, Il'll change it